wsf_scraping/spiders/base_spider.py

```diff
@@ -29,3 +32,17 @@ def _check_headers(self, response_headers,
                        desired_extension=b'application/pdf'):
         content_type = response_headers.get('content-type', '').split(b';')[0]
         return desired_extension == content_type
+
+    def _save_file(self, url, response_body):
+        filename = os.path.basename(urlparse(url).path)
+        tempdir = tempfile.mkdtemp()
+        if filename:
+            if not filename.lower().endswith('.pdf'):
```

Comment:
this is new logic right? where the instances of filenames without .pdf?

Reply:
Some are. All NICE pdf urls aren't ending with `.pdf` for example. For gov.uk, some are and some are not... So I felt it was best to push this logic in the base class

Reply:
👍